### DISCUSSION TITLE
Naming confusion for boost libs?

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#19371] [Q&A]
When we built boost manually against VS2017 (v141) using CMake we got libraries named such as:

> boost_atomic-vc141-mt-gd-x32-1_68.lib

However now we build using vcpkg with VS2019 (v142) we get the library named like:

> boost_atomic-vc140-mt.lib

It seems strange that it is named "v140" and odd that it misses all the extra information from the library like the version number (1.76 IIRC). Is it just the way whoever build the port set it up? Maybe it's considered better the library file keeps the same name across versions?

**Comment by mathisloge** [comment]
Just for reference: 
When i'm building boost with VS2019 and vcpkg I will get `boost_atomic-vc142-mt-x64-1_76.dll`

**Comment by jdx-john** [comment]
Thanks @mathisloge I can confirm the *DLL* seems to have the full name (boost_atomic-vc142-mt-x32-1_76.dll for me) but the .lib does not. 
So I suspect a very minor issue in CMake scripts perhaps?

**Comment by Osyotr** [comment]
See https://github.com/microsoft/vcpkg/commit/b2b2c9136918ed3ef85961c5c2885b9562d07d1d
@ras0219-msft would you kindly elaborate?

**Comment by PhoebeHui** [comment]
See https://github.com/microsoft/vcpkg/issues/601#issuecomment-276220946.

**Reply by jdx-john** [reply]
Why did this get closed without the option to reply? The linked issue does _not_ seem to be a resolution because that is from when VS2017 was released and there was confusion between v140 and v141:

>Boost libraries will be named -vc140 for VS2017 (at least for now)

Now we're on VS2019 v142. In #601 it is discussed that v140 and v141 are guaranteed compatible but this is not the case with v142 IIRC. I ran into an issue in 64 bit builds where there was an incompatibility related to exception handling.

**Reply by PhoebeHui** [reply]
@jdx-john, this is not closed, just moved this to discussions, and please create a discussion instead if you have any questions in future, and we will move all question, vcpkg feature issue to discussion. BTW, we also have a PR to update the issue template.

**Reply by PhoebeHui** [reply]
> I ran into an issue in 64 bit builds where there was an incompatibility related to exception handling.

Could you report an issue for this? it may have a port bug here. I will take a look.

**Reply by jdx-john** [reply]
Thanks for clarifying @PhoebeHui I'm not too familiar with discussion feature. Regarding 64bit, this was not a vcpkg issue. Microsoft changed C++ exception handling model from v141 -> v142, but _only_ for x64. 

It was a while ago but this discusses it: https://devblogs.microsoft.com/cppblog/making-cpp-exception-handling-smaller-x64/
I ran into issues with incompatibility when we migrated to VS2019 on our project, the x64 build stopped working but x86 didn't. So it might be relevant to you guys in terms of compatibility, or it might not be an issue.

**Reply by PhoebeHui** [reply]
cc @ras0219 @ras0219-msft

**Reply by autoantwort** [reply]
So you have an problem because your project is v140 or v141 but you link against the v142 version of the boost lib?

**Reply by mathisloge** [reply]
@autoantwort  the libs are generally named `v140` if a msvc compiler v142> is used
